bdfr/site_downloaders/imgur.py

```diff
@@ -78,7 +78,7 @@ def _get_data(link: str) -> dict:
 
     @staticmethod
     def _validate_extension(extension_suffix: str) -> str:
-        extension_suffix = extension_suffix.strip('?1')
+        extension_suffix = re.sub(r'\?.*', '', extension_suffix)
```

Comment:
Please add tests that show why this is necessary e.g. a test that fails before this change and succeeds afterwards.

Reply:
Outside what I mentioned in my other comment this covers ?2, ?3 and any others that may happen in the future. I'll go back and dig out the on that triggered this to go with the ?1 example and work on a test with them.

Reply:
I've added test cases that should hopefully satisfy this case, Not sure if there's a better way to test it but they verify they work currently.